packages/hardware-trezor/src/transformers/txOut.ts

```diff
@@ -42,17 +42,23 @@ const getScriptHex = (output: Serialization.TransactionOutput): HexBlob | undefi
 
 const getInlineDatum = (datum: Cardano.PlutusData): string => Serialization.PlutusData.fromCore(datum).toCbor();
 
-export const toTxOut: Transform<Cardano.TxOut, Trezor.CardanoOutput, TrezorTxTransformerContext> = (txOut, context) => {
+export const toTxOut: Transform<
+  { txOut: Cardano.TxOut; index: number; isCollateral: boolean },
+  Trezor.CardanoOutput,
+  TrezorTxTransformerContext
+> = (elem, context) => {
+  const { txOut, index, isCollateral } = elem;
   const destination = toDestination(txOut, context);
   const output = Serialization.TransactionOutput.fromCore(txOut);
   const scriptHex = getScriptHex(output);
+  const format = isCollateral ? context?.collateralReturnFormat : context?.outputsFormat[index];
 
-  return context?.useBabbageOutputs
+  return format === Trezor.PROTO.CardanoTxOutputSerializationFormat.MAP_BABBAGE
```

Comment:
Same as before

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1618/commits/3687665ab60d993070a62d2a5d0243ccb99dd330, Trezor target structure reuses more filed so this one could be reduced even more

Reply:
Very nice!